### PR TITLE
[CBRD-22487] Enable wildcard paths in json_search

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-22487

Transformed wildcard json_paths to regexes that match desired json_paths

### CODE REVIEW

File: src/compat/db_json.cpp [hunk 1]
```diff
@@ -2005,6 +2005,76 @@ db_json_remove_func (JSON_DOC &doc, const char *raw_path)
...
+	      ss << "\\.";
+	      break;
+	    case '*':
+	      if (i < wild_card.length() - 1 && wild_card[i + 1] == '*')
+		{
+		  // wild_card '**'. Match any string
+		  ss << "[([:alnum:]|\\.|\\[|\\])]+";
+		  ++i;
+		}
+	      else if (wild_card[i - 1] == '[')
```
Comment (eseokoh) [inline src/compat/db_json.cpp]: it might be safer to add `0 < i && ...`.

**Comment by radudoros** [comment]
@ribram , please confirm you are ok with the PR

File: src/compat/db_json.cpp [hunk 2]
```diff
@@ -2005,6 +2005,76 @@ db_json_remove_func (JSON_DOC &doc, const char *raw_path)
...
+	}
+      ss << "[^[:space:]]*";
+
+      try
+	{
+	  regs.push_back (std::regex (ss.str ()));
+	}
+      catch (std::regex_error &e)
+	{
+	  // regex compilation exception
```
Comment (ribram) [inline src/compat/db_json.cpp]: add an assert (false); safeguard.

File: src/query/string_opfunc.c [hunk 3]
```diff
@@ -3969,46 +3969,89 @@ db_json_search_dbval (DB_VALUE * result, DB_VALUE * args[], const int num_args)
...
-	  return error_code;
-	}
-      return db_make_json (result, result_json, true);
+        {
+          er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OBJ_INVALID_ARGUMENTS, 0);
+          return error_code;
+        }
+    }
+
+  for (size_t i = 0; i<paths.size (); ++i)
```
Comment (ribram) [inline src/query/string_opfunc.c]:
```suggestion
  for (size_t i = 0; i < paths.size (); ++i)
```

File: src/query/string_opfunc.c [hunk 4]
```diff
@@ -3969,46 +3969,89 @@ db_json_search_dbval (DB_VALUE * result, DB_VALUE * args[], const int num_args)
...
+    {
+      for (auto &reg : regs)
+        {
+          matches_path[i] |= (int) std::regex_match (paths[i].substr (1, paths[i].size () - 2), reg);
```
Comment (ribram) [inline src/query/string_opfunc.c]: I propose adapt reg to match paths to avoid extracting sub-strings.

**Pull request merged**